File: sorting/bubble_sort/bubble_sort.ts
function bubbleSort(unsortedArray: number[]): number[] {
    let swapped: boolean;

    let length: number = unsortedArray.length;

    for (let i: number = 0; i < length; i++) {
        swapped = false;
        
        for (let j: number = 0; j < length - i - 1; j ++) {
            if (unsortedArray[j] > unsortedArray[j + 1]) {
                unsortedArray[j], unsortedArray[j + 1] = unsortedArray[j + 1], unsortedArray[j];
                swapped = true;
            }
        }

        if (!swapped) break; 
    }

    return unsortedArray;
}

File: sorting/bubble_sort/BubbleSort.java
public class Sorting {
    public static int[] BubbleSort(int[] unsortedArray) {
        boolean swapped;
        int len = unsortedArray.length();

        for (int i = 0; i < len; i ++) {
            swapped = false;

            for (int j = 0; j < len - i - 1; j ++) {
                if (unsortedArray[j] > unsortedArray[j + 1]) {
                    int temp = unsortedArray[j + 1];
                    unsortedArray[j + 1] = unsortedArray[j];
                    unsortedArray[j] = temp;
                    swapped = true;
                }
            }

            if (!swapped) {
                break;
            }
        }

        return unsortedArray;

    } 
}

class Test {
    public static void main(String[] args) {
        int[] array = {12, 31, 1, 41, 13};

        System.out.println(Sorting.BubbleSort(array));


    }
}